### PR TITLE
attempts to bring supply pod's code up to date + updates adminbuse pod launcher tgui

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

ports
tgstation/tgstation#53833
tgstation/tgstation#52611
and a few more pr-less commits
updates some tgui dependencies (cross might want me to remove this)

## Why It's Good For The Game

https://user-images.githubusercontent.com/56778689/110433118-2d03ad00-8104-11eb-845d-f82f13c910d8.mp4

## Changelog
:cl:jupyterkat, mrdoombringer
admin: admins can now edit the delays for reversing supplypods
admin: The supplypod menu no longer sucks
imageadd: new supplypod icons
imagedel: old supplypod icons
refactor: refactored supplypods code
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Who is helen weinstein?

File: code/modules/cargo/supplypod.dm [hunk 1]
```diff
@@ -124,242 +199,453 @@
...
 		qdel(src) //Delete ourselves and the holder
 		if (holder != src)
 			qdel(holder)
 
-/obj/structure/closet/supplypod/centcompod/close(atom/movable/holder) //Closes the supplypod and sends it back to centcom. Should only ever be called if the "reversing" variable is true
-	handleReturningClose(holder, TRUE)
+/obj/structure/closet/supplypod/close(atom/movable/holder) //Closes the supplypod and sends it back to centcom. Should only ever be called if the "reversing" variable is true
+	if (!holder)
+		return
+	message_admins(holder)
```
Comment (PowerfulBacon) [inline code/modules/cargo/supplypod.dm]: a

File: code/modules/cargo/supplypod.dm [hunk 2]
```diff
@@ -124,242 +199,453 @@
...
+	glow_effect = new(src)
+	glow_effect.icon_state = "pod_glow_" + GLOB.podstyles[style][POD_GLOW]
+	vis_contents += glow_effect
+	glow_effect.layer = GASFIRE_LAYER
+
+/obj/structure/closet/supplypod/proc/endGlow()
+	if(!glow_effect)
+		return
+	glow_effect.layer = LOW_ITEM_LAYER
+	message_admins("HUU!")
```
Comment (PowerfulBacon) [inline code/modules/cargo/supplypod.dm]: b

File: code/modules/cargo/supplypod.dm [hunk 3]
```diff
@@ -124,242 +199,453 @@
...
+	icon_state = "pod_engineglow"
+	desc = ""
+	layer = GASFIRE_LAYER
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	alpha = 255
+
+/obj/effect/engineglow/proc/fadeAway(leaveTime)
+	var/duration = min(leaveTime, 25)
+	animate(src, alpha=0, time = duration)
+	message_admins("HYUU!")
```
Comment (PowerfulBacon) [inline code/modules/cargo/supplypod.dm]: c

**Comment by jupyterkat** [comment]
do note that the shrapnel cloud button doesn’t actually work because we haven’t got the component yet

**Comment by francinum** [comment]
Please add a notice or simply disable the shrapnel cloud button if you do not plan to add the component in this PR.

**Comment by jupyterkat** [comment]
Oh I am planning to add the component in the future

**Review by francinum — CHANGES_REQUESTED** [review]
@jupyterkat Until you do, please grey out the button in the UI, hide it, or otherwise mark it inoperative.

**Comment by jupyterkat** [comment]
button is commented out

**Review by francinum — APPROVED** [review]
Sure that works.

**Comment by francinum** [comment]
No problems after vague abuse. Anything else can get fixed in post.

Approved.

**Pull request merged**